sentry-core/src/api.rs

```diff
@@ -1,44 +1,70 @@
-use crate::breadcrumbs::IntoBreadcrumbs;
-use crate::hub::Hub;
-use crate::internals;
 use crate::protocol::{Event, Level};
-use crate::scope::Scope;
+use crate::types::Uuid;
+use crate::{Hub, IntoBreadcrumbs, Scope};
 
 /// Captures an event on the currently active client if any.
 ///
-/// The event must already be assembled.  Typically code would instead use
-/// the utility methods like `capture_exception`.  The return value is the
-/// event ID.  In case Sentry is disabled the return value will be the nil
-/// UUID (`Uuid::nil`).
+/// The event must already be assembled. Typically code would instead use
+/// the utility methods like [`capture_message`], [`capture_error`], or an
+/// integration specific function.
 ///
-/// # Example
+/// The return value is the event ID. If the event was discarded for any reason,
+/// return value will be the nil UUID (`Uuid::nil`).
+///
+/// # Examples
 ///
 /// ```
-/// # use sentry_core as sentry;
 /// use sentry::protocol::{Event, Level};
+/// use sentry::types::Uuid;
 ///
-/// sentry::capture_event(Event {
+/// let uuid = Uuid::new_v4();
+/// let event = Event {
+///     event_id: uuid,
 ///     message: Some("Hello World!".into()),
 ///     level: Level::Info,
 ///     ..Default::default()
+/// };
+///
+/// assert_eq!(sentry::capture_event(event.clone()), Uuid::nil());
+///
+/// let events = sentry::test::with_captured_events(|| {
+///     assert_eq!(sentry::capture_event(event), uuid);
 /// });
+/// assert_eq!(events.len(), 1);
 /// ```
-pub fn capture_event(event: Event<'static>) -> internals::Uuid {
+///
+/// [`capture_message`]: fn.capture_message.html
```

Comment:
Pretty sure you don't need to link to html anymore, perhaps shouldn't even.
https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html#additions-to-the-documentation-syntax and the next "Implied Shortcut Reference Links" section detail this a bit better.  But basically type a valid rust path to the thing you refer between the square brackets.

Reply:
oh, ignore this.  you've just told me this is nightly-only :(